application/F3DStarter.cxx

```diff
@@ -700,6 +708,123 @@ void F3DStarter::Render()
   f3d::log::debug("Render done");
 }
 
+//----------------------------------------------------------------------------
+void F3DStarter::SaveScreenshot(const std::string& filenameTemplate)
+{
+
+  const auto getScreenshotDir = []()
+  {
+    for (const char* const& candidate : { "XDG_PICTURES_DIR", "HOME", "USERPROFILE" })
+    {
+      char* val = std::getenv(candidate);
+      if (val != nullptr)
+      {
+        std::filesystem::path path(val);
+        if (std::filesystem::is_directory(path))
+        {
+          return path;
+        }
+      }
+    }
+
+    return std::filesystem::current_path();
+  };
+
+  const auto applyFilenameTemplate = [&](const std::string& fn_template)
+  {
+    std::string filename = fn_template;
+
+    {
+      std::regex re("\\{app\\}");
+      filename = std::regex_replace(filename, re, F3D::AppName);
+    }
+    {
+      std::regex re("\\{version\\}");
```

Comment:
I dont see this being used except in the test, am I missing something ?

Reply:
yeah I had it because we had discussed having a special case for "troubleshooting" screenshots where the filename would include the OS and version and the screenshot would have an overlay with more debug info. But we don't have such overlay yet so that part of the feature can't happen yet.
I can remove that part until it's actually needed.

Reply:
to be fair if we go with this semantic, we could do nice things, to be discussed.